scaleway/resource_k8s_cluster.go

```diff
@@ -520,6 +520,7 @@ func resourceScalewayK8SClusterRead(ctx context.Context, d *schema.ResourceData,
 	_ = d.Set("auto_upgrade", clusterAutoUpgradeFlatten(cluster))
 
 	// private_network
+	_ = d.Set("private_network_id", "")
 	if cluster.PrivateNetworkID != nil {
 		_ = d.Set("private_network_id", cluster.PrivateNetworkID)
 	}
```

Comment:
```suggestion
	_ = d.Set("private_network_id", flattenStringPtr(cluster.PrivateNetworkID))
```

You can use `flattenStringPtr`